devlib/target.py

```diff
@@ -152,7 +152,20 @@ def __init__(self,
                  conn_cls=None,
                  ):
         self.connection_settings = connection_settings or {}
-        self.platform = platform or Platform()
+        # Set self.platform either to platform given in connection_settings,
+        if platform is None:
+            self.platform = self.connection_settings('platform', Platform())
```

Comment:
This should be `self.platform = self.connection_settings.get('platform', Platform())`